pkg/neco/cmd/tpm_clear.go

```
@@ -165,19 +164,21 @@ IP is one of the IP addresses owned by the machine.`,
 			machineType := machine.Spec.Labels[machineTypeLabelName]
 			logicType, ok := supportedMachineTypes[machineType]
 			if !ok {
-				return fmt.Errorf("unknown machine type: machine-type=%s", machineType)
+				log.Warn("unknown machine type; shutdown", map[string]interface{}{
+					"serial":       machine.Spec.Serial,
+					"node":         machine.Spec.IPv4[0],
+					"machine_type": machineType,
+				})
+				return power(ctx, "stop", machine.Spec.BMC.IPv4)
```

To power-off anyway would be good.

I prefer to return an error here even when power-off succeeds, because the operator expects "neco tpm clear" both to clear the TPM devices and to power-off the machines.
In this case the TPM devices are not cleared, so not all expectations are fulfilled.
This situation should be expressed by an error rather than a warning.

For QEMU and TPM 1.2 machines, the operator should not expect the command to clear the TPM devices, so returning non-error would be appropriate.

---

That's good. https://github.com/cybozu-go/neco/pull/1807/commits/212012ef740c764f494fd96cafd0853fd60402b3
When logicType for a target machine is not defined, `neco tpm clear` will output the following logs.

```
2021-11-02T02:19:03.222572Z ubuntu-local neco warning: "unknown machine type; shutdown" machine_type="MACHINE_TYPE" node="NNN.NNN.NNN.NNN" result=null serial="SERIAL"
2021-11-02T02:19:03.222647Z ubuntu-local neco error: "unknown machine type"
```